server/src/main/java/io/druid/server/ClientInfoResource.java

```diff
@@ -83,13 +100,83 @@ public ClientInfoResource(
   @Produces(MediaType.APPLICATION_JSON)
   public Map<String, Object> getDatasource(
       @PathParam("dataSourceName") String dataSourceName,
-      @QueryParam("interval") String interval
+      @QueryParam("interval") String interval,
+      @QueryParam("full") String full
   )
   {
-    return ImmutableMap.<String, Object>of(
-        "dimensions", getDatasourceDimensions(dataSourceName, interval),
-        "metrics", getDatasourceMetrics(dataSourceName, interval)
-    );
+    if(full == null) {
```

Comment:
i think that one may be community contributed. We need to enforce style checking on commit and automate the process